connman/src/storage.c

```diff
@@ -794,11 +796,53 @@ static void keyfile_cleanup(void)
 	keyfile_hash = NULL;
 }
 
+static int validate_path(const char *pathname)
+{
+	char resolved_path[PATH_MAX] = { 0 };
+
+	/*
+	 * realpath() expands all symbolic links and references and handles
+	 * NULL pathname.
+	 */
+	if (!realpath(pathname, resolved_path)) {
+		switch (errno) {
+		/*
+		 * File does not exist, which is the case when saving. Verify
+		 * the path to avoid using a broken symlink or unaccessible
+		 * file.
+		 */
+		case ENOENT:
+			DBG("path %s is not readable/has missing symlink (%s)",
+						pathname, resolved_path);
+			break;
+		case EACCES:
+		case EIO:
+		case ELOOP:
+		case ENAMETOOLONG:
+		case ENOMEM:
+		case ENOTDIR:
+		default:
+			DBG("realpath() error %d/%s", errno, strerror(errno));
```

Comment:
Logging also 'pathname' might be useful from debugging point of view.

Reply:
Calling functions do printout a warning in that case when there is an error.

Reply:
ok, so it seems.